server/src/expense/expense.service.ts

```diff
@@ -25,69 +23,103 @@ export class ExpenseService {
+
+        const expenses = await this.prisma.expense.findMany({
+          where: {
+            houseId: userHouse.houseId,
+            date: {
+              gte: firstDayOfMonth,
+              lte: lastDayOfMonth,
+            },
           },
-        },
-        orderBy: {
-          date: 'desc',
-        },
-        include: {
-          payments: {
-            select: {
-              id: true,
-              fee: true,
-              paidDate: true,
-              user: {
-                select: {
-                  id: true,
-                  name: true,
-                  icon: true,
+          orderBy: {
+            date: 'desc',
+          },
+          include: {
+            payments: {
+              select: {
+                id: true,
+                fee: true,
+                paidDate: true,
+                user: {
+                  select: {
+                    id: true,
+                    name: true,
+                    icon: true,
+                  },
                 },
               },
             },
-          },
-          user: {
-            select: {
-              id: true,
-              name: true,
-              icon: true,
+            user: {
+              select: {
+                id: true,
+                name: true,
+                icon: true,
+              },
             },
           },
-        },
-      });
+        });
 
-      const groupedExpenses = expenses.reduce((acc, expense) => {
-        const expenseDate = expense.date.toISOString().split('T')[0];
+        const groupedExpenses = expenses.reduce((acc, expense) => {
+          const expenseDate = expense.date.toISOString().split('T')[0];
 
-        if (!acc[expenseDate]) {
-          acc[expenseDate] = {};
-        }
+          if (!acc[expenseDate]) {
+            acc[expenseDate] = {};
+          }
 
-        const buyerKey = expense.user.id;
+          const buyerKey = expense.user.id;
 
-        if (!acc[expenseDate][buyerKey]) {
-          acc[expenseDate][buyerKey] = [];
-        }
+          if (!acc[expenseDate][buyerKey]) {
+            acc[expenseDate][buyerKey] = [];
+          }
 
-        acc[expenseDate][buyerKey].push(expense);
+          acc[expenseDate][buyerKey].push(expense);
 
-        return acc;
-      }, {});
-      return groupedExpenses;
+          return acc;
+        }, {});
+
+        const expensesByDateArray = Object.keys(groupedExpenses).map((date) => {
+          const expenseDate = new Date(date + 'T00:00:00');
+
+          const monthAbbreviation = expenseDate
+            .toLocaleString('en-US', { month: 'short', timeZone: 'UTC' })
+            .toUpperCase();
+
+          const day = expenseDate.getUTCDate();
+
+          return {
+            date: `${monthAbbreviation} ${day}`,
+            buyers: Object.keys(groupedExpenses[date]).map((buyerId) => ({
+              buyerId,
+              expenses: groupedExpenses[date][buyerId],
+            })),
+          };
+        });
```

Comment:
I think it's better to add a comment here, as it's a little bit complicated! Like you're trying to just add 3 months of data

Reply:
I added comments!
https://github.com/HAK2024/fairy-share/pull/113/commits/bc3704eb1abe8f493af64a16d5e5c220e20d5744